hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsAsyncQueueFactory.kt

```diff
@@ -0,0 +1,121 @@
+package uk.gov.justice.hmpps.sqs
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.boot.actuate.health.HealthIndicator
+import org.springframework.context.ConfigurableApplicationContext
+import software.amazon.awssdk.services.sns.model.SubscribeRequest
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model.CreateQueueRequest
+import software.amazon.awssdk.services.sqs.model.GetQueueAttributesRequest
+import software.amazon.awssdk.services.sqs.model.GetQueueUrlRequest
+import software.amazon.awssdk.services.sqs.model.GetQueueUrlResponse
+import software.amazon.awssdk.services.sqs.model.QueueAttributeName
+import uk.gov.justice.hmpps.sqs.HmppsSqsProperties.QueueConfig
+
+class HmppsAsyncQueueFactory(
+  private val context: ConfigurableApplicationContext,
+  private val sqsAsyncClientFactory: SqsAsyncClientFactory,
+) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun createHmppsAsyncQueues(hmppsSqsProperties: HmppsSqsProperties, hmppsTopics: List<HmppsAsyncTopic> = listOf()) =
```

Comment:
very minor, but suspect it should just be `createHmppsQueues` since there is no such thing as a async queue

Reply:
Hmmm interesting point. The HmppsQueue is just a vessel to hold the queue/DLQ names and SQS clients. Therefore HmppsAsyncQueue is the same but holds the async SQS clients. But as you say an async queue doesn't mean anything 🤔